### PR TITLE
cmd: Added support for downloading to another DB instance

### PR DESCRIPTION
**Pull request by Arachnid**

### CODE REVIEW

File: cmd/geth/chaincmd.go [hunk 1]
```diff
@@ -268,6 +285,181 @@ func exportChain(ctx *cli.Context) error {
...
+	if err := dl.RegisterPeer("local", 63, peer); err != nil {
+		return err
+	}
+
+	currentHeader := hc.CurrentHeader()
+	if err := dl.Synchronise("local", currentHeader.Hash(), hc.GetTd(currentHeader.Hash(), currentHeader.Number.Uint64()), syncmode); err != nil {
+		return err
+	}
+	for dl.Synchronising() {
+		time.Sleep(1000)
```
Comment (fjl) [inline cmd/geth/chaincmd.go]: That's 1000ns. Please add a unit, e.g. `1 * time.Microsecond`. Maybe it should sleep longer, too.

Comment (fjl) [inline cmd/geth/chaincmd.go]: That's 1000ns. Please add a unit, e.g. `1 * time.Microsecond`. Maybe it should sleep longer, too.

Comment (fjl) [inline cmd/geth/chaincmd.go]: That's 1000ns. Please add a unit, e.g. `1 * time.Microsecond`. Maybe it should sleep longer, too.

**Comment by GitCop** [comment]
Thank you for your contribution! Your commits seem to not adhere to the repository coding standards

 * Commit: f4af32ea9c03b0856385974d0be0e64114d84b4f
  * Commits must be prefixed with the package(s) they modify

Please check the [contribution guidelines](https://github.com/ethereum/go-ethereum#contribution) for more details.

---
This message was auto-generated by https://gitcop.com

File: cmd/geth/chaincmd.go [hunk 2]
```diff
@@ -91,6 +94,21 @@ Optional second and third arguments control the first and
...
+	copydbCommand = cli.Command{
+		Action:    utils.MigrateFlags(copyDb),
+		Name:      "copydb",
+		Usage:     "Copy from one chain DB into another using the downloader",
+		ArgsUsage: "<filename>",
+		Flags: []cli.Flag{
+			utils.DataDirFlag,
+			utils.CacheFlag,
+			utils.SyncModeFlag,
+			utils.FakePoWFlag,
```
Comment (karalabe) [inline cmd/geth/chaincmd.go]: I think we're going to need 

```
			utils.TestnetFlag,
			utils.RinkebyFlag,
```

too to allow copying over the testnet chains.

File: cmd/geth/chaincmd.go [hunk 3]
```diff
@@ -91,6 +94,21 @@ Optional second and third arguments control the first and
...
+		},
+		Category: "BLOCKCHAIN COMMANDS",
+		Description: `
+The first argument must be the directory containing the blockchain to download from`,
```
Comment (karalabe) [inline cmd/geth/chaincmd.go]: From, or to? Aren't we downloading from the `--datadir`?

**Comment by karalabe** [comment]
I've moved the localPeer into `downloader.FakePeer` so it may be reused by other code too if need be (also it was an annoyingly large chunk of code inside `cmd/geth/chaincmd.go`). The last commit also fixes a small issue where PoA chains couldn't be copied due to `utils.MakeChain` always picking ethash.

@Arachnid @fjl @holiman PTAL

**Comment by karalabe** [comment]
It's LGTM from me and @Arachnid. But since it's a combo PR, @fjl or @holiman please thumb it up once again.

**Pull request merged**